packages/flutter/lib/src/rendering/object.dart

```diff
@@ -4751,726 +4501,1163 @@ mixin RelayoutWhenSystemFontsChangeMixin on RenderObject {
-      showOnScreen: owner.showOnScreen,
-      owner: owner.owner!.semanticsOwner!,
-    );
-    final SemanticsNode node = owner._semantics!;
-    assert(MatrixUtils.matrixEquals(node.transform, Matrix4.identity()));
-    assert(node.parentSemanticsClipRect == null);
-    assert(node.parentPaintClipRect == null);
+  _RenderObjectSemantics get owner;
 
-    node.rect = owner.semanticBounds;
+  bool mergesToSibling = false;
 
-    final List<SemanticsNode> children = <SemanticsNode>[];
-    for (final _InterestingSemanticsFragment fragment in _children) {
-      assert(fragment.config == null);
-      fragment.compileChildren(
-        parentSemanticsClipRect: parentSemanticsClipRect,
-        parentPaintClipRect: parentPaintClipRect,
-        elevationAdjustment: 0.0,
-        result: children,
-        siblingNodes: siblingNodes,
-      );
-    }
-    // Root node does not have a parent and thus can't attach sibling nodes.
-    assert(siblingNodes.isEmpty);
-    node.updateWith(config: null, childrenInInversePaintOrder: children);
+  void markSiblingConfigurationConflict(bool conflict);
+}
 
-    // The root node is the only semantics node allowed to be invisible. This
-    // can happen when the canvas the app is drawn on has a size of 0 by 0
-    // pixel. If this happens, the root node must not have any children (because
-    // these would be invisible as well and are therefore excluded from the
-    // tree).
-    assert(!node.isInvisible || children.isEmpty);
-    result.add(node);
-  }
+/// A fragment that is generated from
+/// [SemanticsConfiguration.childConfigurationsDelegate]
+///
+/// A render object can choose to add additional semantics config to be merged
+/// upward besides itself. These configs can be added through
+/// [SemanticsConfiguration.childConfigurationsDelegate] and will form
+/// `_IncompleteSemanticsFragment`s
+///
+/// See [RenderParagraph] for an example usage.
+class _IncompleteSemanticsFragment extends _SemanticsFragment {
+  _IncompleteSemanticsFragment(this.configToMergeUp, this.owner);
 
   @override
-  SemanticsConfiguration? get config => null;
-
-  final List<_InterestingSemanticsFragment> _children = <_InterestingSemanticsFragment>[];
+  final SemanticsConfiguration configToMergeUp;
 
   @override
-  void markAsExplicit() {
-    // nothing to do, we are always explicit.
-  }
+  final _RenderObjectSemantics owner;
 
   @override
-  void addAll(Iterable<_InterestingSemanticsFragment> fragments) {
-    _children.addAll(fragments);
+  void markSiblingConfigurationConflict(bool conflict) {
+    assert(!conflict);
   }
 }
 
-/// A fragment with partial information that must not form an explicit
-/// semantics node without merging into another _SwitchableSemanticsFragment.
+/// A wrapper class for a [RenderObject] that provides semantics related
+/// properties and compilations.
+///
+/// The _RenderObjectSemantics tree is compiled in three phases.
+///
+/// *Phase 1*
```

Comment:
Is this an accurate high-level description? 

In Phase 1 & 2 (executed by updateChildren) we build/update a tree of _RenderObjectSemantics by querying dirty RenderObjects about their SemanticsConfiguration and updating the _RenderObjectSemantics of the render objects according to these information. In this phase we also decide what _RenderObjectSemantics will have to be their own SemanticsNodes later on. In phase 3 (executed by ensureSemanticsNode) we are then compiling these _RenderObjectSemantics into the actual SemanticsNodes that form the semantics tree.

Maybe add a high-level description like this to the doc?